cli/src/main/kotlin/com/wire/kalium/cli/ConversationsApplication.kt

```diff
@@ -19,15 +20,22 @@ class ConversationsApplication : CliktCommand() {
         val loginResult = networkModule.loginApi.emailLogin(
             LoginWithEmailRequest(email = email, password = password, label = "ktor"),
             false
-        ).resultBody
-
-        credentialsLedger.onAuthenticate(loginResult.accessToken, "") //TODO extract refresh token from cookie response
-
-        val conversations = networkModule.conversationApi.conversationsByBatch(null, 100).resultBody.conversations
-
-        println("Your conversations:")
-        conversations.forEach {
-            println("ID:${it.id}, Name: ${it.name}")
+        )
+
+        if (!loginResult.isSuccessful()) {
+            println("There was an error on the login :( check the credentials and the internet connection and try again please")
+        } else {
+            credentialsLedger.onAuthenticate(loginResult.value.accessToken, "") //TODO extract refresh token from cookie response
```

Comment:
This is only a comment, not a request for change.

This piece of code reminded me of one extra possibility using the contracts in `isSuccessful()`: Guard clauses.

```kotlin
if(!loginResult.isSuccessful()) {
    println("Failure to authenticate")
    return
}
loginResult.value.accessToken // The compiler allows it! If it was an error, this code would be reachable
```

We don't need to nest checks 🤩

Reply:
Yes I saw that. I nested it because I didn't want to return / halt the execution of the application, just in case we came up later with a mechanism of retry the login in case it fails. But can remove the if else if you prefer it that way. Not very opinionated tbh

Reply:
Not very opinionated for this one too. Just wanted to highlight that we do have this possibility going forward.

I'll mark this as resolved :)